server/gitrest/src/utils/isomorphicgitConversions.ts

```diff
@@ -0,0 +1,205 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import * as isomorphicGit from "isomorphic-git";
+import * as resources from "@fluidframework/gitresources";
+import { NetworkError } from "@fluidframework/server-services-client";
+
+type IsomorphicGitTreeEntryType = "commit" | "blob" | "tree";
+type IsomorphicGitTagObjectType = IsomorphicGitTreeEntryType | "tag";
+
+function ensureIsomorphicGitTreeEntryType(originalITreeEntryType: string): IsomorphicGitTreeEntryType {
+    if (!["commit", "blob", "tree"].includes(originalITreeEntryType)) {
+        throw new NetworkError(400, "Invalid TreeEntry type.");
+    }
+
+    return originalITreeEntryType as IsomorphicGitTreeEntryType;
+}
+
+function ensureIsomorphicGitTagObjectType(originalITagType: string): IsomorphicGitTagObjectType {
+    if (!["commit", "blob", "tree", "tag"].includes(originalITagType)) {
+        throw new NetworkError(400, "Invalid Tag Object type.");
+    }
+
+    return originalITagType as IsomorphicGitTagObjectType;
+}
+
+function oidToCommitHash(oid: string): resources.ICommitHash {
+    return { sha: oid, url: "" };
+}
+
+/**
+ * Helper function to convert an `isomorphic-git` ReadCommitResult to our resource representation
+ */
+export function commitToICommit(commitResult: isomorphicGit.ReadCommitResult): resources.ICommit {
+    return {
+        author: {
+            date: new Date(commitResult.commit.author.timestamp * 1000).toISOString(),
+            email: commitResult.commit.author.email,
+            name: commitResult.commit.author.name,
+        },
+        committer: {
+            date: new Date(commitResult.commit.committer.timestamp * 1000).toISOString(),
+            email: commitResult.commit.committer.email,
+            name: commitResult.commit.committer.name,
+        },
```

Comment:
(nit) use a function to avoid repeating the code?

Reply:
Do you mean there is repetition when populating `author` and `committer`? The code is actually not repeated. `author` is using `commit.author`. `committer` is using `commit.committer`. Or maybe you meant something else?

Reply:
I think there are a couple places where we get `something.commit.author.name/email`. Perhaps `getAuthorNameEmailFromCommit(commit: ??): { name: string, email: string } { ...}`? Maybe not worth it for just 3 repeats though